lua/ttt2/libraries/database.lua

```diff
@@ -393,310 +411,165 @@ end
+-- Should make sure, that it doesnt go to infinity
 -- @realm client
 -- @internal
-clientSendFunctions[MESSAGE_REGISTER] = function(data)
-	net.WriteUInt(data, uIntBits)
+local function receivedIdentifier()
+	identifiersReceived = identifiersReceived + 1
+
+	if identifiersReceived < messageIdentifier then return end
+
+	identifiersReceived = 0
+	messageIdentifier = 0
 end
 
 ---
 -- Receive query for registered databases from client
 -- and sync them
+-- @param table data = {identifier}
 -- @param string plyID64 the playerID64 of the player who sent the message
 -- @realm server
 -- @internal
-serverReceiveFunctions[MESSAGE_REGISTER] = function(plyID64)
-	database.SyncRegisteredDatabases(plyID64, net.ReadUInt(uIntBits))
-end
-
----
--- Send requested registered databases to client
--- @param table data contains identifier, tableCount, index here
--- @realm server
--- @internal
-serverSendFunctions[MESSAGE_REGISTER] = function(data)
-	local databaseInfo = registeredDatabases[data.index]
-
-	net.WriteUInt(data.index, uIntBits)
-	net.WriteString(databaseInfo.accessName)
-	net.WriteTable(databaseInfo.keys)
-	net.WriteTable(databaseInfo.data)
-
-	-- AdditionalInfo determines if there is more than one database registered and sent
-	-- This makes sure, that the client doesnt start to use the databases before all databases are received
-	local sendAdditionalInfo = data.identifier ~= nil
-
-	net.WriteBool(sendAdditionalInfo)
-
-	if sendAdditionalInfo then
-		net.WriteUInt(data.identifier, uIntBits)
-		net.WriteUInt(data.tableCount, uIntBits)
-	end
+serverReceiveFunctions[MESSAGE_REGISTER] = function(data, plyID64)
+	database.SyncRegisteredDatabases(plyID64, data.identifier)
 end
 
 ---
 -- Receive requested registered databases from server
 -- and cache them as well as call all cached functions, that are waiting for the databases
+-- @param table data = {index, accessName, savingKeys, additionalData, sentAdditionalInfo, identifier, tableCount}
 -- @realm client
 -- @internal
-clientReceiveFunctions[MESSAGE_REGISTER] = function()
-	local index = net.ReadUInt(uIntBits)
-	local accessName = net.ReadString()
-	local savingKeys = net.ReadTable()
-	local additionalData = net.ReadTable()
-
-	nameToIndex[accessName] = index
-	registeredDatabases[index] = {
-		accessName = accessName,
-		keys = savingKeys,
-		data = additionalData,
+clientReceiveFunctions[MESSAGE_REGISTER] = function(data)
+	nameToIndex[data.accessName] = data.index
+	registeredDatabases[data.index] = {
+		accessName = data.accessName,
+		keys = data.savingKeys,
+		data = data.additionalData,
 		storedData = {},
 		defaultData = {}
 	}
 
-	-- If more than one database will be send, then sentAdditionalInfo is true
-	local sentAdditionalInfo = net.ReadBool()
-
-	if sentAdditionalInfo then
-		local identifier = net.ReadUInt(uIntBits)
-		local tableCount = net.ReadUInt(uIntBits)
-
+	if data.identifier and table.Count(registeredDatabases) == data.tableCount then
```

Comment:
Just a note: Maybe the variable name `data.tableCount` is a bit confusing. At least I first thought you were misusing a function reference.

maybe something like `data.sizeTable`?

Reply:
Well then I always prefer real words. So either tablesize or size I'd like better than sizeTable. Because to mention its a table would be kinda unneccessary.

Reply:
Maybe someone else wants to chime in as well, but if a variable is named `tableSize` I expect a table with the size. Hence my suggestion of `sizeTable`. I don't know if this is just me or a general coding style. However datatypes are generally put in front like `boolState` for example, sometimes just `bState`

Reply:
I can see what you mean but many variables are called:
screenSize, iconSize, (ClipSize, scope_size), itemSize, queueSize, scrollSize
and I only find:
sizeIcon, which contradicts iconSize you use inside the drawing functions
heightButton, widthButton
and SizeVel which is also not camelCase.

So Im personally no fan of combined words, that sound strange.
and bState, well is a boolean, but we seldom actually use it.
In c++ we also rather have a diffrence between private m_state and public variables state.

Normally I would call it like
numberOfTables or databaseSize could be fine if you really dont like tableSize.

Reply:
We actually use tableSize in table.lua and sh_rolelayering.lua.

Reply:
yes, I started that change just recently, because I stubled over the variable names like the one in this comment. However it is fine for me either way, just a preference thing